src/loaders/workspace_downloader/workspace_downloader.py

```diff
@@ -0,0 +1,356 @@
+            _list_objects_params(wsid, min_id, max_id, filter_objects_name_by)
+        )
+        for min_id, max_id in batch_input
+    ]
+    res_objs = list(itertools.chain.from_iterable(objs))
+    return res_objs
+
+
+def process_input(conf):
+    """
+    Download .fa and .meta files from workspace and save a copy under output_dir.
+    """
+    while True:
+        task = conf.queue.get(block=True)
+        if not task:
+            print("Stopping")
+            break
+        upa, obj_info = task
+
+        # cfn points to the assembly file outside of the container
+        # get_assembly_as_fasta writes the file to /kb/module/workdir/tmp/<filename> inside the container.
+        # workdir is shared between the container and the external file system
+        # Any file path get_assembly_as_fasta returns will be relative to inside the container, and so is not useful for this script
+
+        cfn = os.path.join(conf.job_dir, "workdir/tmp", upa)
+        # upa file is downloaded to cfn
+        conf.asu.get_assembly_as_fasta({"ref": upa.replace("_", "/"), "filename": upa})
+
+        # each upa in output_dir as a seperate directory
+        dstd = os.path.join(conf.pth, upa)
+        os.makedirs(dstd, exist_ok=True)
+
+        dst = os.path.join(dstd, f"{upa}.fa")
+        # Hard link .fa file from job_dir to output_dir in WS
+        os.link(cfn, dst)
+
+        metafile = os.path.join(dstd, f"{upa}.meta")
+        # save meta file with relevant object_info
+        with open(metafile, "w", encoding="utf8") as json_file:
+            json.dump(_process_object_info(obj_info), json_file, indent=2)
+
+        print("Completed %s" % (upa))
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="PROTOTYPE - Download genome files from the workspace service (WSS)."
+    )
+
+    required = parser.add_argument_group("required named arguments")
+    optional = parser.add_argument_group("optional arguments")
+
+    # Required flag argument
+    required.add_argument(
+        "--workspace_id",
+        required=True,
+        type=int,
+        help="Workspace addressed by the permanent ID",
+    )
+
+    # Optional argument
+    optional.add_argument(
+        "--root_dir",
+        type=str,
+        default=loader_common_names.ROOT_DIR,
+        help="Root directory.",
+    )
+    optional.add_argument(
+        "--kb_base_url",
+        type=str,
+        default=loader_common_names.KB_BASE_URL,
+        help="KBase base URL, defaulting to prod",
+    )
+    optional.add_argument(
+        "--workers",
+        type=int,
+        default=5,
+        choices=range(1, cpu_count() + 1),
+        help="Number of workers for multiprocessing",
+    )
+    optional.add_argument(
+        "--token_filename",
+        type=str,
+        help="Filename in home directory that stores token",
+    )
+    optional.add_argument(
+        "--delete_job_dir",
+        action="store_true",
+        help="Delete job directory",
+    )
```

Comment:
Does this mean by default the job dir doesn't get deleted? If so that should be reversed

Reply:
Updated!

Reply:
:+1: